azurerm/resource_arm_application_gateway.go

```diff
@@ -2158,7 +2409,17 @@ func flattenApplicationGatewayURLPathMaps(input *[]network.ApplicationGatewayURL
 						ruleOutput["name"] = *rule.Name
 					}
 
+					// TODO: Note below: ruleProps versus props - bug?
 					if ruleProps := rule.ApplicationGatewayPathRulePropertiesFormat; props != nil {
+
+						if ruleProps.BackendAddressPool != nil && ruleProps.RedirectConfiguration != nil {
+							return nil, fmt.Errorf("[ERROR] Conflict between `backend_address_pool_name` and `redirect_configuration_name` (back-end pool not applicable when redirection specified)")
+						}
+
+						if ruleProps.BackendHTTPSettings != nil && ruleProps.RedirectConfiguration != nil {
```

Comment:
(same here) could we check != nil && != ""?

Reply:
Done